src/main/java/no/fdk/dataservicecatalog/service/parser/Parser.java

```diff
@@ -24,26 +27,41 @@ enum ApiType {
     ApiSpecification parse(String spec) throws ParseException;
 
     static boolean isValidSwaggerOrOpenApiV3(String spec, ApiType apiType, String minimalVersion) {
-        try {
-            JsonElement element = new JsonParser().parse(spec);
-            String version = element.getAsJsonObject().get(apiType.label).getAsString();
-            if (!(version.length() > 2 && version.startsWith(minimalVersion))) {
+        OpenAPIMeta specMeta = readMandatoryMetaProperties(spec);
+        if (specMeta != null) {
+            String version;
+            if (apiType == ApiType.OPENAPI) {
+                version = specMeta.getOpenapi();
+            } else {
+                version = specMeta.getSwagger();
+            }
+            if (version == null || !(version.length() > 2 && version.startsWith(minimalVersion))) {
```

Comment:
Er dette enklere å lese?
```suggestion
            if (version == null || version.length() < 3 || !version.startsWith(minimalVersion)) {
```

Reply:
Når parser sier at at den takler openapi spec med "minimalVersion" = 3 så sjekker dette feltet `openapi: 3.0.2` øverst i speccen

Kanskje litt mer leselig, Det er uansett en fil jeg kommer til å slette i neste pr, så gjorde minimalt til endringer her.

Den sjekker at swagger/openapi feltet har en verdi og at den starter med oppgitt major-versjon. Lengde-sjekken virker strengt tatt ganske snodig, men jeg kommer som sagt til å slette fila...

Reply:
cb94b10

Reply:
For å ha noe i stedet for lengdesjekken som gir mer mening: 7bebb08

Men er usikker på hvilke krav openapi-gjengen egentlig har til versjoneringen sin. La her til at patch er optional siden vi har eksempel på det i testfilene, så regner med at det er ok å gjøre irl